scripts/verify-deployment.ts

```diff
@@ -0,0 +1,64 @@
+import { execSync } from 'child_process';
+import yargs from 'yargs';
+
+type ProcessArgs = {
+  version: string;
+  isPrerelease: boolean;
+};
+const githubBinaries = ['amplify-pkg-linux-arm64.tgz', 'amplify-pkg-linux.tgz', 'amplify-pkg-macos.tgz', 'amplify-pkg-win.exe.tgz'];
+const parseArgs = async (): Promise<ProcessArgs> => {
+  const args = yargs(process.argv.slice(2))
+    .options({
+      v: { type: 'string', demandOption: true },
+      p: { type: 'boolean', default: false },
+    })
+    .parseSync();
+
+  return { version: args.v, isPrerelease: args.p };
+};
+
+const existsInNpm = (version: string): boolean => {
+  try {
+    execSync(`npm show @aws-amplify/cli@${version}`, {
+      stdio: 'ignore',
+    });
+    return true;
+  } catch {
+    return false;
+  }
+};
+const existsInGitHub = async (version: string): Promise<boolean> => {
+  const buildUrl = (binary: string) => `https://github.com/aws-amplify/amplify-cli/releases/download/v${version}/${binary}`;
+  const responses = await Promise.all(githubBinaries.map((binary) => fetch(buildUrl(binary), { method: 'HEAD' })));
+  const responseStatuses = responses.map((r) => r.status);
+  return responseStatuses.every((s) => s === 200);
+};
+
+const s3Binaries = ['amplify-pkg-linux-arm64.tgz', 'amplify-pkg-linux-x64.tgz', 'amplify-pkg-macos-x64.tgz', 'amplify-pkg-win-x64.tgz'];
+const existsInS3 = async (version: string): Promise<boolean> => {
+  const buildUrl = (binary: string) => `https://package.cli.amplify.aws/${version}/${binary}`;
+  const responses = await Promise.all(s3Binaries.map((b) => fetch(buildUrl(b))));
+  const responseStatuses = responses.map((r) => r.status);
+  return responseStatuses.every((s) => s === 200);
+};
+
+const main = async () => {
+  const { version, isPrerelease } = await parseArgs();
+  console.log(`#### Verifying version ${version} deployed correctly ####`);
+  const isGitHubSatisfied = isPrerelease || (await existsInGitHub(version));
```

Comment:
pre-release would have binaries too no ?
look at https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/21287/workflows/ab5552e3-1bee-43ce-9406-1f54738d914e 
We upload binaries , then craft pre-release, ship to npm and promote pre-release to latest. All share same binaries and version number.

The `existsInGitHub` should take `isPrerelease` flag and check status of release (i.e. is latest or no)

Reply:
Prerelease may not be the correct word here. I'm referring to RCs and tagged releases with this. Those do not get uploaded to GitHub.